### PR TITLE
Add unmanaged Puppet Agent runmode

### PR DESCRIPTION
**Pull request by Rathios**
Fixes GH-383

### CODE REVIEW

**Comment by ekohl** [comment]
How does this differ from runmode `none`?

**Comment by Rathios** [comment]
Left a comment here too: https://github.com/theforeman/puppet-puppet/issues/383

Bottom line is I would like to be able to leave the service stopped while testing changes, undisturbed by the service. In that context I would also like to be able to enable the service without starting it, which I can do with a profile in the control-repository, but it's not currently possible to manually manage the service yourself as long as this module is used to install the agent.

File: manifests/agent/service/cron.pp [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class puppet::agent::service::cron (
   Boolean $enabled = false,
 ) {
-  unless 'cron' in $::puppet::unavailable_runmodes {
+  unless 'unmanaged' in $::puppet::runmode or 'cron' in $::puppet::unavailable_runmodes {
```
Comment (ekohl) [inline manifests/agent/service/cron.pp]: runmode is a string, so you can compare it with `$::puppet::runmode == 'unmanaged'`

**Comment by mmoll** [comment]
I'm OK with it after fixing :+1:

**Comment by Rathios** [comment]
Ha, you're right. The consistency OCD missed the fact that $::puppet::unavailable_runmodes is NOT a string. Things make sense now. Fixed.

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**